### PR TITLE
Redesign inventory, stats and configuration page to use tabs

### LINKED ISSUE #7088: Redesign stats page to use tabs
## Description
Redesign stats page to use tabs

image

### PR DESCRIPTION
**Pull request by guidomodarelli**
### Description
Redesign inventory, stats, and configuration page to use tabs.

Remove unused styles.

We want to remove the agent name, and add three tabs:

Software, Network, Processes

We want to move tables to the appropriate tabs:

Software: Packages, Windows Updates
Network: Network interfaces, Network ports, Network settings
Processes: Processes
 
### Issues Resolved
Closes #7088, #7108, #7111

### Evidence

image

image

image

image

image

image

### Test
1. Go to `Agents management` and click on `Summary`.
2. Choose an Agent from the list.
3. Review the redesigned `inventory`, `statistics`, and `configuration` page for tab usage.

In `inventory Data`:

  We want to remove the agent name, and add three tabs:
  
  `Software`, `Network`, `Processes`
  
  We want to move tables to the appropriate tabs:
  
  - `Software`: `Packages`, `Windows Updates`
  - `Network`: `Network interfaces`, `Network ports`, `Network settings`
  - `Processes`: `Processes`
  
  We want to maintain the ribbon in all three tabs
  
- The `ribbon` must have the same design as the dashboard one

#### Unit tests

> `public/components/common/modules/main-agent.test.tsx`
> 
> image

> public/components/agents/syscollector/inventory.test.tsx
>
> image

> public/components/agents/stats/agent-stats.test.tsx
>
> image

> `public/controllers/management/components/management/configuration/configuration-switch.test.tsx`
>
> image

> `public/components/agents/syscollector/components/syscollector-metrics.test.tsx`
>
> image

### Check List
- [x] All tests pass
  - [x] `yarn test:jest`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [x] Update [CHANGELOG.md](./../CHANGELOG.md)
- [x] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -6,12 +6,11 @@
   "editor.inlineSuggest.enabled": true,
   "editor.insertSpaces": true,
   "editor.minimap.enabled": true,
-  "editor.rulers": [80, 100],
+  "editor.rulers": [110],
```
Comment (Desvelao) [inline .vscode/settings.json:9]: **question**: why is this changed? It will cause the files will format and we will have changes related to code styling instead of the intentional change.

File: plugins/main/public/components/common/modules/main-agent.tsx [hunk 2]
```diff
@@ -66,179 +73,124 @@ export class MainModuleAgent extends Component {
...
+                    onClick={() => this.props.switchTab?.(AgentTabs.NETWORK)}
+                  >
+                    {toTitleCase(AgentTabs.NETWORK)}&nbsp;
+                  </EuiTab>
+                  <EuiTab
+                    isSelected={this.props.section === AgentTabs.PROCESSES}
+                    onClick={() => this.props.switchTab?.(AgentTabs.PROCESSES)}
+                  >
+                    {toTitleCase(AgentTabs.PROCESSES)}&nbsp;
+                  </EuiTab>
```
Comment (Desvelao) [inline plugins/main/public/components/common/modules/main-agent.tsx:101]: **nitpick**: this could be refactored using and array and mapping the render.

**Review by Desvelao — CHANGES_REQUESTED** [review]
Revert the changes in `.prettierrc` nad `.vscode.json` settings and run the code formatter in the changed files of the pull request.

Some test failed related to snapshots.

File: plugins/main/public/components/common/modules/main-agent.tsx (hunk 2 above)
Comment (guidomodarelli) [reply plugins/main/public/components/common/modules/main-agent.tsx]: Fixed

**Comment by Desvelao** [comment]
# :green_circle: Test

image
image
image

**Comment by JuanGarriuz** [comment]
### Test
1. Go to Server Management and click on 'Endpoints Summary'.
2. Choose an Agent from the list.
3. Review the redesigned inventory, statistics, and configuration page for tab usage.


- Stats:

image

- Configuration

image

- Inventory:

image




In `inventory Data`:

  We want to remove the agent name, and add three tabs:
  
  `Software`, `Network`, `Processes`
  
  
image

  
  
  We want to move tables to the appropriate tabs:
  
  - `Software`: `Packages`, `Windows Updates`
  
  
image

  
  - `Network`: `Network interfaces`, `Network ports`, `Network settings`
  
  
image

  
  - `Processes`: `Processes`
  
  
image

  
  We want to maintain the ribbon in all three tabs
  
- The `ribbon` must have the same design as the dashboard one

**Comment by guidomodarelli** [comment]
:green_circle: Server Management > Settings

https://github.com/user-attachments/assets/098063c9-a732-402f-827c-113c8e615ae1

**Comment by guidomodarelli** [comment]
:green_circle: Configuration
> Groups are displayed in the ribbon and another panel

image

**Review by chantal-kelm — APPROVED** [review]
CR: ✅ 
TEST: ✅ 

Captura de pantalla 2024-10-23 a la(s) 4 53 34 p  m

Captura de pantalla 2024-10-23 a la(s) 4 53 05 p  m

Captura de pantalla 2024-10-23 a la(s) 4 46 49 p  m

Captura de pantalla 2024-10-23 a la(s) 4 55 04 p  m

Captura de pantalla 2024-10-23 a la(s) 4 55 07 p  m

Captura de pantalla 2024-10-23 a la(s) 4 55 22 p  m

**Review by chantal-kelm — APPROVED** [review]
CR: ✅ 
TEST: ✅ 

Captura de pantalla 2024-10-23 a la(s) 4 55 22 p  m
Captura de pantalla 2024-10-23 a la(s) 4 55 07 p  m
Captura de pantalla 2024-10-23 a la(s) 4 55 04 p  m
Captura de pantalla 2024-10-23 a la(s) 4 53 34 p  m
Captura de pantalla 2024-10-23 a la(s) 4 53 05 p  m
Captura de pantalla 2024-10-23 a la(s) 4 46 49 p  m

File: plugins/main/public/components/common/ribbon/ribbon-item.tsx [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+  const wzWidth100 = { anchorClassName: 'wz-width-100' };
+  const tooltipProps = item.label === 'Cluster node' ? wzWidth100 : {};
+
+  const getPlatformIcon = (agent?: Agent) => {
+    let icon = '';
+    const { uname, platform } = agent?.os || {};
+
+    if (
+      uname?.toLowerCase().includes(WAZUH_AGENTS_OS_TYPE.LINUX) ||
+      platform?.toLowerCase().includes('ubuntu')
```
Comment (Desvelao) [inline plugins/main/public/components/common/ribbon/ribbon-item.tsx:63]: **question**: why is this neccesary? We used the another logic to know if the host is Linux. If your addition for Ubuntu is required, then this should take into account in other views too.

Comment (guidomodarelli) [reply plugins/main/public/components/common/ribbon/ribbon-item.tsx:63]: I can change for `sysname`
image

File: plugins/main/public/components/common/welcome/agents-welcome.js [hunk 4]
```diff
@@ -326,14 +326,17 @@ export const AgentsWelcome = compose(
...
+                style={{ marginTop: 7, marginRight: '0.5rem' }}
+              >
                 <ButtonExploreAgent />
               </EuiFlexItem>
               <EuiFlexItem grow={false} style={{ marginTop: 7 }}>
                 <WzButton
                   buttonType='empty'
                   iconType='inspect'
-                  onClick={() => this.props.switchTab('syscollector')}
+                  onClick={() => this.props.switchTab(AgentTabs.SOFTWARE)}
```
Comment (Desvelao) [inline plugins/main/public/components/common/welcome/agents-welcome.js:339]: **question**: why the value of the tab was renamed from `syscollector` to `software`?

**Review by asteriscos — APPROVED** [review]
CR: :green_circle: 
Test: :red_circle: 


https://github.com/user-attachments/assets/0e3f1e3f-66a7-48bd-9379-4b52b65fb6ca

The information displayed of the same agent changes. It says the status in unknown, but it is known.

Base branch
image

Current branch
image

File: plugins/main/public/components/endpoints-summary/agent/index.tsx [hunk 5]
```diff
@@ -105,17 +108,41 @@ export const AgentView = compose(
...
-      <Route path='?tab=stats'>
+      <Route path={`?tab=${AgentTabs.STATS}`}>
         <MainModuleAgent agent={agentData} section={tab} />
         <MainAgentStats agent={agentData} />
       </Route>
-      <Route path='?tab=configuration'>
+      <Route path={`?tab=${AgentTabs.CONFIGURATION}`}>
         <MainModuleAgent agent={agentData} section={tab} />
-        <WzManagementConfiguration agent={agentData} />
+        <WzManagementConfiguration showAgentInfo agent={agentData} />
```
Comment (Desvelao) [inline plugins/main/public/components/endpoints-summary/agent/index.tsx:145]: **question**: what does the `showAgentInfo` prop?

File: plugins/main/public/components/common/welcome/agents-welcome.js (hunk 4 above)
Comment (guidomodarelli) [reply plugins/main/public/components/common/welcome/agents-welcome.js:339]: Since the `syscollector` tab no longer exists, I have selected the first tab to be displayed when you click on `Inventory Data`.

File: plugins/main/public/components/endpoints-summary/agent/index.tsx (hunk 5 above)
Comment (guidomodarelli) [reply plugins/main/public/components/endpoints-summary/agent/index.tsx]: I forgot to delete it.

**Comment by guidomodarelli** [comment]
> The information displayed of the same agent changes. It says the status in unknown, but it is known.
> 
> Base branch image
> 
> Current branch image

image

**Review by asteriscos — APPROVED** [review]
CR: :green_circle: 
Test: :green_circle: 

Peek 2024-10-28 18-09

**Comment by Desvelao** [comment]
# :green_circle: Test

image
image
image
image
image

**Review by Desvelao — APPROVED** [review]
LGTM!
